internal/zinc-compile-core/src/main/scala/sbt/internal/inc/AnalyzingCompiler.scala

```diff
@@ -125,74 +122,14 @@ final class AnalyzingCompiler(
     ()
   }
 
-  def newCachedCompiler(
+  override def newCachedCompiler(
       arguments: Array[String],
       output: Output,
       log: xLogger,
       reporter: Reporter
-  ): CachedCompiler =
-    newCachedCompiler(arguments: Seq[String], output, log, reporter)
-
-  def newCachedCompiler(
-      arguments: Seq[String],
-      output: Output,
-      log: xLogger,
-      reporter: Reporter
-  ): CachedCompiler = {
-    val (bridge, bridgeClass) = bridgeInstance(compilerBridgeClassName, log)
-    val compiler = bridge match {
-      case intf: CompilerInterface2 =>
-        intf.newCompiler(arguments.toArray[String], output, log, reporter)
-      case _ =>
-        // fall back to old reflection if CompilerInterface1 is not supported
-        invoke(bridge, bridgeClass, "newCompiler", log)(
-          classOf[Array[String]],
-          classOf[Output],
-          classOf[xLogger],
-          classOf[Reporter]
-        )(arguments.toArray[String], output, log, reporter)
-    }
-    compiler.asInstanceOf[CachedCompiler]
-  }
-
-  def doc(
-      sources: Seq[VirtualFile],
-      classpath: Seq[VirtualFile],
-      outputDirectory: Path,
-      options: Seq[String],
-      maximumErrors: Int,
-      log: ManagedLogger
-  ): Unit = {
-    val reporter = new ManagedLoggedReporter(maximumErrors, log)
-    doc(sources, classpath, outputDirectory, options, log, reporter)
-  }
```

Comment:
Why is doc being deleted?

Reply:
Unused.

Reply:
Used by sbt I think - https://github.com/sbt/sbt/blob/develop/main-actions/src/main/scala/sbt/Doc.scala